bug: The the required CUTLASS submodule is not download during build

### Describe the bug

```
make[1]: Entering directory '/root/vllm/semantic-router'
==================> Running rust ============> ...
Loading Rust environment from /root/.cargo/env...
Building Rust library with CUDA and Flash Attention support (ENABLE_FLASH_ATTN=1)...
✅ nvcc found: Cuda compilation tools, release 12.4, V12.4.131
   Note: Flash Attention requires CUDA Compute Capability >= 8.0 (RTX 3090+, A100, H100)
   Compiling candle-flash-attn v0.9.2-alpha.1
error: failed to run custom build command for `candle-flash-attn v0.9.2-alpha.1`

Caused by:
  process didn't exit successfully: `/root/vllm/semantic-router/candle-binding/target/release/build/candle-flash-attn-7dfdd271d9ecc013/build-script-build` (exit status: 101)
  --- stdout
...
  #$ gcc -std=c++17 -D__CUDA_ARCH_LIST__=890 -D__NV_LEGACY_LAUNCH -E -x c++ -D__CUDACC__ -D__NVCC__ -D__CUDACC_EXTENDED_LAMBDA__ -D__CUDACC_RELAXED_CONSTEXPR__  -fPIC -O3 -I"cutlass/include" "-I/usr/local/cuda/bin/../targets/x86_64-linux/include"    -U "__CUDA_NO_HALF_OPERATORS__" -U "__CUDA_NO_HALF_CONVERSIONS__" -U "__CUDA_NO_HALF2_OPERATORS__" -U "__CUDA_NO_BFLOAT16_CONVERSIONS__" -D__CUDACC_VER_MAJOR__=12 -D__CUDACC_VER_MINOR__=4 -D__CUDACC_VER_BUILD__=131 -D__CUDA_API_VER_MAJOR__=12 -D__CUDA_API_VER_MINOR__=4 -DCUDA_API_PER_THREAD_DEFAULT_STREAM=1 -D__NVCC_DIAG_PRAGMA_SUPPORT__=1 -include "cuda_runtime.h" -m64 "kernels/flash_fwd_hdim224_bf16_sm80.cu" -o "/tmp/tmpxft_00005708_00000000-5_flash_fwd_hdim224_bf16_sm80.cpp4.ii"
  In file included from kernels/flash_fwd_launch_template.h:12,
                   from kernels/flash_fwd_hdim96_fp16_causal_sm80.cu:5:
  kernels/flash_fwd_kernel.h:9:10: fatal error: cute/tensor.hpp: No such file or directory
      9 | #include <cute/tensor.hpp>
        |          ^~~~~~~~~~~~~~~~~
  compilation terminated.
  In file included from kernels/flash_fwd_launch_template.h:12,
                   from kernels/flash_fwd_hdim96_bf16_sm80.cu:5:
  kernels/flash_fwd_kernel.h:9:10: fatal error: cute/tensor.hpp: No such file or directory
      9 | #include <cute/tensor.hpp>
        |          ^~~~~~~~~~~~~~~~~
  compilation terminated.
  # --error 0x1 --

  thread '<unnamed>' (22265) panicked at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/bindgen_cuda-0.1.5/src/lib.rs:262:21:
  nvcc error while executing compiling: "nvcc" "--gpu-architecture=sm_89" "-c" "-o" "/root/vllm/semantic-router/candle-binding/target/release/build/candle-flash-attn-1d4202f8d2c4ea62/out/flash_fwd_hdim96_fp16_causal_sm80-779d1cffdf403141.o" "--default-stream" "per-thread" "-std=c++17" "-O3" "-U__CUDA_NO_HALF_OPERATORS__" "-U__CUDA_NO_HALF_CONVERSIONS__" "-U__CUDA_NO_HALF2_OPERATORS__" "-U__CUDA_NO_BFLOAT16_CONVERSIONS__" "-Icutlass/include" "--expt-relaxed-constexpr" "--expt-extended-lambda" "--use_fast_math" "--verbose" "-Xcompiler" "-fPIC" "kernels/flash_fwd_hdim96_fp16_causal_sm80.cu"
...
```

### To Reproduce

run `make rust ENABLE_FLASH_ATTN=1`on gpu machine

### Expected behavior

_No response_

### Additional context

_No response_

--- a/candle-binding/Cargo.toml
+++ b/candle-binding/Cargo.toml
@@ -21,12 +21,13 @@ flash-attn = ["cuda", "candle-flash-attn"]
 
 [dependencies]
 anyhow = { version = "1", features = ["backtrace"] }
-candle-core = "0.9.2-alpha.1"
-candle-nn = "0.9.2-alpha.1"
-candle-transformers = "0.9.2-alpha.1"
+candle-core = { git = "https://github.com/huggingface/candle", tag = "0.9.2-alpha.1" }
+candle-nn = { git = "https://github.com/huggingface/candle", tag = "0.9.2-alpha.1" }
+candle-transformers = { git = "https://github.com/huggingface/candle", tag = "0.9.2-alpha.1" }
 # Flash Attention 2 (optional, requires CUDA)
 # Reference: https://github.com/huggingface/candle/tree/main/candle-flash-attn
-candle-flash-attn = { version = "0.9.2-alpha.1", optional = true }
+# Using git dependency to automatically fetch CUTLASS submodule
+candle-flash-attn = { git = "https://github.com/huggingface/candle", tag = "0.9.2-alpha.1", optional = true }
 tokenizers = { version = "0.21.0", features = ["http"] }
 hf-hub = "0.4.1"
 safetensors = "0.4.1"

